casper/src/main/resources/Registry.rho

```diff
@@ -43,9 +291,13 @@ in {
       new seed, uriCh in {
         ops!("buildUri", *seed.toByteArray(), *uriCh) |
         for (@uri <- uriCh) {
-          for (@map <- _registryStore) {
-            _registryStore!(map.set(uri, data)) |
-            ret!(uri)
+          for (@map <<- _registryStore) {
```

Comment:
Let's assume that there are two inserts that run in parallel where both inserts share the same unpopulated common prefix with a key that is already in the TreeHashMap.
In that case we could loose one insert due to using peek here.
The same applies to `insertSigned`.

Reply:
No, the peek here just gets the root of the tree.  Once we get to a leaf, the code that updates the map at the leaf, `nodeSetMap`, acquires a lock on the map at the leaf before adding the key.